test/test_faiss.py

```diff
@@ -66,20 +62,40 @@ def test_faiss_update_docs(document_store, index_buffer_size):
     documents_indexed = document_store.get_all_documents()
 
     # test if number of documents is correct
-    assert len(documents_indexed) == len(documents)
+    assert len(documents_indexed) == len(DOCUMENTS)
 
     # test if two docs have same vector_is assigned
     vector_ids = set()
     for i, doc in enumerate(documents_indexed):
         vector_ids.add(doc.meta["vector_id"])
-    assert len(vector_ids) == len(documents)
+    assert len(vector_ids) == len(DOCUMENTS)
 
     # test if correct vectors are associated with docs
     for i, doc in enumerate(documents_indexed):
-        original_doc = [d for d in documents if d["text"] == doc.text][0]
+        original_doc = [d for d in DOCUMENTS if d["text"] == doc.text][0]
         updated_embedding = retriever.embed_passages([Document.from_dict(original_doc)])
         stored_emb = document_store.faiss_index.reconstruct(int(doc.meta["vector_id"]))
         # compare original input vec with stored one (ignore extra dim added by hnsw)
-        assert np.allclose(updated_embedding, stored_emb[:-1])
+        assert np.allclose(updated_embedding, stored_emb[:-1], rtol=0.01)
+
+@pytest.mark.parametrize("document_store", ["faiss"], indirect=True)
+def test_faiss_retrieving(document_store):
+
+    document_store.write_documents(DOCUMENTS)
+
+    retriever = EmbeddingRetriever(document_store=document_store, embedding_model="deepset/sentence_bert", use_gpu=False)
+    embedding = retriever.retrieve(query="How to test this?")
```

Comment:
Should we add some assertions for the embedding? This test would pass even if embedding is `None`.

Reply:
Done